### ISSUE TITLE
Document local development and referencing of generators in cookbook

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#53468]
The cookbook doesn't document how to do local development and referencing of source generators in a single project. Think we should add that in as it's a typical case, particularly now that we support debugging such scenarios as of VS 16.10.

```xml
<ProjectReference 
  Include="..\SourceGeneratorSamples\SourceGeneratorSamples.csproj"
  OutputItemType="Analyzer"
  ReferenceOutputAssembly="false" />
```

**Comment by sharwell** [comment]
I marked a feedback ticket as a duplicate of this issue:
https://developercommunity.visualstudio.com/t/Analyzer-satellite-libraries-not-copied-/10737439

It's not identical because it's describing an analyzer instead of a source generator, but from an infrastructure view these features would behave the same way.

**Comment by zorgoz** [comment]
Hi @sharwell.

Yes, using those attributes works indeed. But then there are still caching issues. Both CLI-build and from-IDE-build result in no diagnostic results from the analyzer, but the code editor still shows the errors from before the analyzer code was changed.

Not even restarting VS IDE helps. Not until the cached analyzer is deleted from the location I have noted in my original report:  `%TEMP%\VBCSCompiler\AnalyzerAssemblyLoader`. The problem is, that this folder is locked by an instance of `dotnet.exe` and one `VBCSCompiler.exe` even after the VS IDE is closed. I had no time to wait them out to exit, even though I had waited quite a lot.
